jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -68,10 +68,16 @@ private void add(TarArchiveEntry tarArchiveEntry) {
         return;
       }
 
+      Path namePath = Paths.get(tarArchiveEntry.getName());
+
+      // Don't add entries for "/"
+      if (namePath.equals(Paths.get("/"))) {
+        return;
+      }
+
       // Adds all directories along extraction paths to explicitly set permissions for those
       // directories.
-      Path namePath = Paths.get(tarArchiveEntry.getName());
-      if (namePath.getParent() != namePath.getRoot()) {
+      if (namePath.getParent() != null) {
```

Comment:
It turns out this was only ever true if `getParent` and `getRoot` both returned null.
1. This was doing an object based equality anyway
2. `getRoot` is ALWAYS null, because these paths have their leading slash stripped by the TarArchiveEntry
3. we handle the only possible non-null `getRoot` entry above at: https://github.com/GoogleContainerTools/jib/pull/2300/files#diff-9f3ce3d4c2254dbe95c91624e7b434d9R74

Reply:
This is contingent on us never creating TarArchiveEntries that are `preserveAbsolutePath == true`.

I think our previous code would work the same way anyway.

Reply:
> This was doing an object based equality anyway

This is bad. However, I got a deja-vu having been suspicious about this check and thought perhaps this code actually works. Indeed, the following code did return true on my machine.
```java
    Path namePath = Paths.get("/foo");
    System.out.println(namePath.getRoot() == namePath.getParent());
```
I can guess why, but we should have really used `equals()`.

> * `getRoot` is ALWAYS null, because these paths have their leading slash stripped by the TarArchiveEntry
> * we handle the only possible non-null `getRoot` entry above at:

Isn't this claim in conflict with the check above (`namePath.equals(Paths.get("/")`)? When the check is true, `namePath.getRoot()` will return `/`.

Anyways, I guess comparing with `getRoot()` is general and future proof regardless of `preserveAbsolutePath`?

Reply:
Well, can't call equals because of null pointer exception. This whole section is super suspecious, and I'm actually not too confident about this part of code. I think I'd rather push to 2.2.0 milestone than submit here.

Reply:
The claim is that when TarArchiveEntry returns name, it's stripped of any "/" prefixed to the path, so the path is never absolute, so root cannot be determined.

So if you do
```
TarArchiveEntry tae = new TarArchiveEntry(new File("some/file"), "/foo");
System.out.println(tae.getName())
```

you will get `foo` and not `/foo`

unless you have explicitly done
```
TarArchiveEntry tae = new TarArchiveEntry("/foo", true)
System.out.println(tae.getName())
```

Reply:
> The claim is that when TarArchiveEntry returns name, it's stripped of any "/" prefixed to the path

This is true. What I'm saying is that this is in conflict with the following check you added:
```java
      // Don't add entries for "/"
      if (namePath.equals(Paths.get("/"))) {
```
Because `new TarArchiveEntry("/").getName()` is an empty string (i.e, `namePath` is `Paths.get("")` which is differetn from `Paths.get("/")`), it seems to be that the check above is meaningless.

Reply:
It seems like the only unclear part is whether to check for "/" or "". Arguably both would represent entries that shouldn't go into the archive.
Why not check for both and get this forward?
```
      // Don't add entries for "/"
      if (tarArchiveEntry.getName().equals("")
       || tarArchiveEntry.getName().equals("/")) {
        return;
      }
```
Also, no need to convert Strings to Path for this part. Only adds another indirection when we already know exactly which strings we're looking for.

Reply:
I thought about this for a bit. I think `if (tarArchiveEntry.getName().equals("/")` could work. I tried this condition out with the test we have in the PR and it seems to give us the desired outcome. However, the `tarArchiveEntry.getName().equals("")` check may not be needed. Testing this out through the debugger shows that the tarArchiveEntry is created at this [line](https://github.com/GoogleContainerTools/jib/blob/71e3977d5b1e81b860b29c3c62c8ea807a1d2d28/jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java#L152) which results in "/" when .getName() is called for the extraction path of "/".

Reply:
I don't remember all the details about the context of this problem, but what I said back then may be relevant: https://github.com/GoogleContainerTools/jib/pull/2300/files#r381532140

> Ah, I see the difference. If you use `TarArchiveEntry(File)`, `getName()` returns `"/"`. However, when you use `TarArchiveEntry(String)`, it returns `""`.

Reply:
Thanks @chanseokoh! It's interesting that they behave differently. That's definitely helpful here.